File: src/gamesbyexample/guess.py
"""Guess the Number, by Al Sweigart al@inventwithpython.com
Try to guess the secret number based on hints.
This code is available at https://nostarch.com/big-book-small-python-programming
Tags: tiny, beginner, game"""
__version__ = 0
import random


def askForGuess():
    while True:
        guess = input('> ')  # Enter the guess.

        if guess.isdecimal():
            return int(guess)  # Convert string guess to an integer.
        print('Please enter a number between 1 and 100.')


print('Guess the Number, by Al Sweigart al@inventwithpython.com')
print()
secretNumber = random.randint(1, 100)  # Select a random number.
print('I am thinking of a number between 1 and 100.')

for i in range(10):  # Give the player 10 guesses.
    print('You have {} guesses left. Take a guess.'.format(10 - i))

    guess = askForGuess()
    if guess == secretNumber:
        break  # Break out of the for loop if the guess is correct.

    # Offer a hint:
    if guess < secretNumber:
        print('Your guess is too low.')
    if guess > secretNumber:
        print('Your guess is too high.')

# Reveal the results:
if guess == secretNumber:
    print('Yay! You guessed my number!')
else:
    print('Game over. The number I was thinking of was', secretNumber)

File: computercraft/guess.lua
--[[Guess the Number
by Al Sweigart al@inventwithpython.com
Try to guess the secret number based on hints.
]]

function askForGuess()
  while true do
    local guess = read() -- Enter the guess.

    if tonumber(guess) ~= nil then
      return tonumber(guess)
    end
    print('Enter a number between 1 and 100.')
  end
end

print('Guess the Number')
print('By Al Sweigart al@inventwithpython.com')
local secretNumber = math.random(1, 100)
print('I am thinking of a number between 1 and 100.')

for i = 0, 9 do -- Give the player 10 guesses.
  print('You have ' .. (10 - i) .. ' guesses left.')
  print('Take a guess.')

  local guess = askForGuess()
  if guess == secretNumber then
    break
  end

  -- Offer a hint:
  if guess < secretNumber then
    print('Your guess is too low.')
  end
  if guess > secretNumber then
    print('Your guess is too high.')
  end
end

-- Reveal the results:
if guess == secretNumber then
  print('Yay! You guessed my number!')
else
  print('Game over. The number was ' .. secretNumber)
end
